lib/puppet/configurer/fact_handler.rb

```diff
@@ -32,6 +32,6 @@ def facts_for_uploading
 
     text = facts.render(:pson)
 
-    {:facts_format => :pson, :facts => Puppet::Util.uri_query_encode(text)}
```

Comment:
The problem reported in https://tickets.puppetlabs.com/browse/PUP-7658 is caused by this line in particular.  `uri_query_encode` helper was changed so that it conceptually works over an entire query string - and unencodes any `&` that may have been encoded.  In the case here, all of the fact data should be encoded, else it injects delimiters into values that end up being URI percent encoded OR form-encoded.  The correct helper to use here is actually [URI.encode_www_form_component(str)](https://ruby-doc.org/stdlib-1.9.3/libdoc/uri/rdoc/URI.html#method-c-encode_www_form_component)